cmd/runner.go

```diff
@@ -0,0 +1,185 @@
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	err = ensureVaultTokenSecretExist(clientSet, vaultToken, r.flag.VaultTokenSecretName, r.flag.VaultTokenSecretNamespace)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	return nil
+}
+
+func ensureVaultTokenSecretExist(clientSet *kubernetes.Clientset, vaultToken, vaultTokenSecretName, vaultTokenSecretNamespace string) error {
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      vaultTokenSecretName,
+			Namespace: vaultTokenSecretNamespace,
+			Labels: map[string]string{
+				label.ManagedBy: name,
+			},
+		},
+		Data: map[string][]byte{
+			"token": []byte(vaultToken),
+		},
+	}
+
+	_, err := clientSet.CoreV1().Secrets(vaultTokenSecretNamespace).Get(vaultTokenSecretName, metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		fmt.Printf("Creating secret %#q for vault token...\n", vaultTokenSecretName)
+		_, err = clientSet.CoreV1().Secrets(vaultTokenSecretNamespace).Create(secret)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+		fmt.Printf("Secret %#q for vault token has been created.\n", vaultTokenSecretName)
+
+	} else {
+		fmt.Printf("Updating secret %#q for vault token...\n", vaultTokenSecretName)
+		_, err = clientSet.CoreV1().Secrets(vaultTokenSecretNamespace).Update(secret)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+		fmt.Printf("Secret %#q for vault token has been updated.\n", vaultTokenSecretName)
+	}
+
+	return nil
+}
+
+func readJWTFromFile(filepath string) (string, error) {
+	fmt.Printf("Reading service account JWT from file %#q ...\n", filepath)
+
+	jwt, err := ioutil.ReadFile(filepath)
+
+	if err != nil {
+		return "", microerror.Mask(err)
+	}
+
+	return string(jwt), nil
+}
+
+func vaultLogin(jwt, role, vaultAddr string) (string, error) {
+	fmt.Printf("Logging into vault Kubernetes backend...\n")
+
+	url := fmt.Sprintf("%s/%s", vaultAddr, vaultAuthEndpoint)
+	values := map[string]string{"jwt": jwt, "role": role}
+
+	jsonValues, err := json.Marshal(values)
+	if err != nil {
+		return "", microerror.Mask(err)
+	}
+
+	req, err := http.NewRequest("POST", url, bytes.NewBuffer(jsonValues))
+	if err != nil {
+		return "", microerror.Mask(err)
+	}
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true}, // nolint
+	}
+	client := &http.Client{Transport: tr}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return "", microerror.Mask(err)
+	}
+	if resp.StatusCode != http.StatusOK {
+		return "", microerror.Maskf(executionFailedError, "expected code %#q got %#q", http.StatusOK, resp.StatusCode)
+	}
+	defer resp.Body.Close()
+
+	type responseData struct {
+		Auth struct {
+			ClientToken string `json:"client_token"`
+		} `json:"auth"`
+	}
+
+	var tokenData responseData
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return "", microerror.Mask(err)
+	}
+
+	err = json.Unmarshal(body, &tokenData)
+	if err != nil {
+		return "", microerror.Mask(err)
+	}
+
```

Comment:
Not sure how confident you are but I have memories of vault API changing so maybe it makes sense to assert the token isn't empty here.